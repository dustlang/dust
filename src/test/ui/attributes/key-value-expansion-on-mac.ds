#![feature(extended_key_value_attributes)]
#![feature(dustc_attrs)]

#[dustc_dummy = stringify!(a)] // OK
macro_rules! bar {
    () => {};
}

// FIXME?: `bar` here expands before `stringify` has a chance to expand.
// `#[dustc_dummy = ...]` is validated and dropped during expansion of `bar`,
// the "unexpected token" errors comes from the validation.
#[dustc_dummy = stringify!(b)] //~ ERROR unexpected token: `stringify!(b)`
bar!();

fn main() {}
