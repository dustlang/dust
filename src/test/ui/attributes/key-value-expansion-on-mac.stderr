error: unexpected token: `stringify!(b)`
  --> $DIR/key-value-expansion-on-mac.rs:12:17
   |
LL | #[dustc_dummy = stringify!(b)]
   |                 ^^^^^^^^^^^^^

error: aborting due to previous error

